### PR TITLE
Update digest and sha2 versions to 0.10

### PR DESCRIPTION
**Pull request by remoun**
Also update `bincode`, `criterion` and `hex`, but those are minor.

Supports https://github.com/mobilecoinfoundation/mobilecoin/issues/1326

### CODE REVIEW

**Review by cbeck88 — DISMISSED** [review]
lgtm

**Review by cbeck88 — APPROVED** [review]
LGTM: it would be great to get this in upstream though, maybe we can talk to Isis?

**Review by jcape — APPROVED** [review]
LGTM, agree with Chris that it's worth trying to get these changes upstream

File: Cargo.toml [hunk 1]
```diff
@@ -31,10 +31,10 @@ features = ["nightly", "simd_backend"]
...
 
 [dev-dependencies]
-sha2 = { version = "0.9", default-features = false }
-bincode = "1"
-criterion = "0.3.0"
-hex = "0.4.2"
+sha2 = { version = "0.10", default-features = false }
+bincode = "1.3.3"
+criterion = "0.3.5"
+hex = "0.4.3"
```
Comment (isislovecruft) [inline Cargo.toml:37]: These are all semver equivalent to my knowledge.

**Review by isislovecruft — COMMENTED** [review]
This pull should be made against dalek-cryptography:curve25519-dalek, and then that upstream should be merged here.

File: Cargo.toml (hunk 1 above)
Comment (isislovecruft) [reply Cargo.toml:37]: Oh sorry, not the sha2 change, I don't know why that got included in the diff, just the bincode, criterion, and hex changes.

Comment (remoun) [reply Cargo.toml:37]: I was thinking I'd update the MCF forks in line with what we've been doing, then send another PR merging into upstraem dalek-crypto:master. Would that work?

**Pull request merged**